### PR TITLE
not allowed to delete system-generated token

### PR DESCRIPTION
**Pull request by iamigo**

### CODE REVIEW

**Comment by shriramshankar** [comment]
Just for my understanding, how do we know if a token is system generated or not ?

**Comment by iamigo** [comment]
system-generated token has token name = user name

File: tests/db/model/token/delete.js [hunk 1]
```diff
@@ -40,6 +40,28 @@ describe('db: Token: find', () => {
...
+      t = token;
+      done();
+    })
+    .catch(done);
+  });
+  afterEach(u.forceDelete);
+  it('Not allowed to delete system token', (done) => {
+    Token.findOne({ where: { name: t.name } })
+    .then((token) => token.destroy())
+    .then(() => done('expecting error here'))
```
Comment (annyhe) [inline tests/db/model/token/delete.js]: Why do we expect an error here? The user hasn't been deleted yet.

Comment (iamigo) [inline tests/db/model/token/delete.js]: We don't want to let a user delete the system-generated token for their user record, so the token.destroy call should send back an error

**Pull request merged**